src/platform/qt/library/LibraryTree.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2014-2017 waddlesplash
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#pragma once
+
+#include <QTreeWidget>
+
+#include "LibraryController.h"
+
+// Predefinitions
+namespace Ui { class LibraryTree; }
+
+namespace QGBA {
+
+class LibraryTree : public QTreeWidget, public AbstractGameList {
+Q_OBJECT
+
+public:
+	explicit LibraryTree(QWidget* parent_widget = nullptr);
+	~LibraryTree();
+
+	// AbstractGameList stuff
+	virtual LibraryEntry selectedEntry() override;
+	virtual void selectEntry(LibraryEntry game) override;
+
+	virtual void setViewStyle(LibraryStyle newStyle) override;
+
+	virtual void addEntries(QList<LibraryEntry> items) override;
+	virtual void addEntry(LibraryEntry item) override;
+	virtual void removeEntry(LibraryEntry item) override;
+
+signals:
+	void StartGame();
+
+private slots:
+	void ItemActivated(QTreeWidgetItem* item);
+
+private:
+	enum Columns {
+		COL_NAME = 0,
+		COL_LOCATION = 1,
+		COL_PLATFORM = 2,
+		COL_SIZE = 3,
+		COL_CRC32 = 4,
+	};
+
+	Ui::LibraryTree* m_ui;
+	LibraryStyle m_currentStyle;
+
+	bool m_deferredTreeRebuild = false;
+	QMap<QTreeWidgetItem*, LibraryEntry> m_items;
+	QMap<QString, QTreeWidgetItem*> m_path_nodes;
```

Comment:
`m_pathNodes`. Any chance of these being managed pointers so you don't have to manually delete each one?